src/pages/Vendors.tsx

```diff
@@ -0,0 +1,258 @@
+import ActionsMenu from '@/components/features/ActionsMenu'
+import SkeletonPage from '@/components/features/SkeletonPage'
+import DataTable from '@/components/shared/DataTable'
+import FormDialog from '@/components/shared/FormDialog'
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { TextField, Checkbox, FormControlLabel } from '@mui/material'
+import { Controller, UseFormReturn, SubmitHandler } from 'react-hook-form'
+// import { SubmitHandler } from 'react-hook-form'
+
+import { VendorFormData, vendorSchema } from '@/schemas/vendorSchema'
+
+export default function Vendors() {
+  const { t: translate } = useTranslation()
+  const [openDialog, setOpenDialog] = useState(false)
+  const [isFinal, setIsFinal] = useState(false)
+
+  const handleClickOpen = () => {
+    setOpenDialog(true)
+  }
+
+  const onCloseDialog = () => {
+    setOpenDialog(false)
+  }
+
+  const handleSubmit: SubmitHandler<VendorFormData> = (data) => {
+    const formData = {
+      ...data,
+      isFinal
+    }
+    console.log(formData)
+    onCloseDialog()
+  }
+
+  function renderForm({ control, formState: { errors } }: UseFormReturn<VendorFormData>) {
```

Comment:
rename to something like `CreateVendorForm`

Reply:
Fixed